### PR TITLE
[JENKINS-47101] Implement ArgumentsActionImpl with EnvironmentExpander

### PR DESCRIPTION
**Pull request by car-roll**
Remove the safe list from `ArgumentsActionImpl` and replace with the enhanced `EnvironmentExpander` from #370 
This should remove the need for a safe list that can cause incorrect application of masked variables.

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImplTest.java [hunk 1]
```diff
@@ -156,18 +156,20 @@ public void testStringSafetyTest() throws Exception {
...
+
+        Assert.assertFalse("EnvVars that do not occur are safe", ArgumentsActionImpl.isStringSensitive("I have no passwords", new EnvVars(passwordBinding), sensitiveVariables));
+
+//        HashMap<String, String> safeBinding = new HashMap<>();
+//        safeBinding.put("harmless", "secret");
+//        HashSet<String> safeVars = new HashSet<>();
+//        safeVars.add("harmless");
+//        passwordBinding.put("harmless", "secret");
+//        Assert.assertTrue("Input containing whitelisted bound value is safe", ArgumentsActionImpl.isStringSafe(input, new EnvVars(safeBinding), safeVars));
+//        Assert.assertFalse("Input containing one safe and one unsafe bound value is unsafe", ArgumentsActionImpl.isStringSafe(input, new EnvVars(passwordBinding), safeVars));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImplTest.java:172]: If this is obsolete, just delete it, do not comment it out.

**Review by jglick — COMMENTED** [review]
IIUC this subsumes #370 and just adds https://github.com/jenkinsci/workflow-cps-plugin/pull/378/commits/6e8b5eb99200853905b7eedb6c91ec7678600d42?

File: src/test/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImplTest.java (hunk 1 above)
Comment (car-roll) [reply src/test/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImplTest.java:172]: yeah, i meant to, just forgot to do that in the clean up.

**Comment by car-roll** [comment]
@jglick sorry for the confusion. I should have just made this a PR to #370 
#370 is still going through review, but I just wanted to take a first stab at doing this  to see how much work would need to be done. I'm still inclined to keep these two separate PRs.

**Comment by jglick** [comment]
> I should have just made this a PR to #370

Perhaps. There are downsides to that too. Unfortunately GH does not provide first-class support for “stacked” PRs. For example, you can display the incremental diff as https://github.com/car-roll/workflow-cps-plugin/compare/interpolation-v2...remove-safe-list but then reviewers cannot add line comments. You can display the incremental diff in the **Files** tab here but you cannot create a permalink to that as the heads of one or both PRs move.

**Comment by car-roll** [comment]
Closing this PR and merging with #370

**Pull request closed without merging**